src/modules/sway/layout.cpp

```diff
@@ -0,0 +1,101 @@
+#include "modules/sway/layout.hpp"
+#include <spdlog/spdlog.h>
+
+namespace waybar::modules::sway {
+
+Layout::Layout(const std::string& id, const Json::Value& config)
+    : ALabel(config, "layout", id, "{}", 0) {
+  ipc_.subscribe(R"(["input"])");
+  ipc_.signal_event.connect(sigc::mem_fun(*this, &Layout::onEvent));
+  ipc_.signal_cmd.connect(sigc::mem_fun(*this, &Layout::onCmd));
+  ipc_.sendCmd(IPC_GET_INPUTS);
+  shortName();
+  // Launch worker
+  worker();
+  dp.emit();
+}
+
+void Layout::onEvent(const struct Ipc::ipc_response& res) {
+  try {
+    std::lock_guard<std::mutex> lock(mutex_);
+    auto                        payload = parser_.parse(res.payload);
+    if (payload["change"] == "xkb_layout") {
+      layout_ = payload["input"]["xkb_active_layout_name"].asString();
+    }
+    dp.emit();
+  } catch (const std::exception& e) {
+    spdlog::error("Layout: {}", e.what());
+  }
+}
+
+void Layout::onCmd(const struct Ipc::ipc_response &res) {
+  if (res.type == IPC_GET_INPUTS) {
+    try {
+      std::lock_guard<std::mutex> lock(mutex_);
+      auto                        payload = parser_.parse(res.payload);
+      for (auto keyboard : payload) {
+        if (keyboard["identifier"] == "1:1:AT_Translated_Set_2_keyboard") {
```

Comment:
Are you sure that everyone have the same keyboard identifier?
Can you make a config for that with this value as default :)

Reply:
It is most common, but yes it is not same for everyone.

Another option it is to show active layout from any input device. To my knowledge all devices have at startup the same keyboard layout.

Or we can load this identifier from sway config.

Which option do you think is the best?

I like your idea with config but maybe as default use just first keyboard in list?

Reply:
I suppose both solution can be combined together. Allowing to specify an identifier in the configuration, or defaulting to the first keyboard.

Using the first keyboard is both a terrible and good assumption because :
 - a lot of devices are considered keyboard. e.g. :
```
❯ swaymsg -t get_inputs 
Input device: HP Wireless hotkeys
  Type: Keyboard
  Identifier: 0:0:HP_Wireless_hotkeys
  Product ID: 0
  Vendor ID: 0
  Active Keyboard Layout: English (US)
  Libinput Send Events: enabled

Input device: HP WMI hotkeys
  Type: Keyboard
  Identifier: 0:0:HP_WMI_hotkeys
  Product ID: 0
  Vendor ID: 0
  Active Keyboard Layout: English (US)
  Libinput Send Events: enabled

...
```
 - but I guess we can assume that everybody will either use `input * xkb_layout <layout>` and `input * xkb_switch_layout <index>` (which also change the layout for my wireless button for example), or manually specify their keyboard identifier.